src/goto-instrument/contracts/contracts.cpp

```diff
@@ -753,10 +752,10 @@ void code_contractst::instrument_call_statement(
 
       // Make freshly allocated memory assignable, if we can determine its type.
       assigns_clause_targett *new_target =
-        assigns_clause.add_pointer_target(rhs);
+        assigns_clause.add_target(dereference_exprt(rhs));
       goto_programt &pointer_capture = new_target->get_init_block();
 
-      int lines_to_iterate = pointer_capture.instructions.size();
+      size_t lines_to_iterate = pointer_capture.instructions.size();
       program.insert_before_swap(local_instruction_iterator, pointer_capture);
       std::advance(instruction_iterator, lines_to_iterate + 1);
```

Comment:
If you rebase your branch on to the current `develop`, you wouldn't need to keep track of `lines_to_iterate` any more. I added a helper function for this, and you would be able to do:

```c
insert_before_swap_and_advance(program, instruction_iterator, pointer_capture);
```

I don't understand the `+ 1` on the `lines_to_iterate` though. Why is it needed?

Reply:
`+1` wasn't necessary there, it came from this case:
```c
    // malloc statments return a void pointer, which is then cast and assigned
    // to a result variable. We iterate one line forward to grab the result of
    // the malloc once it is cast.
```
I re-wrote the code to use `insert_before_swap_and_advance`. Good catch!

Reply:
Awesome; thanks!